### PR TITLE
Checks for duplicate paths in store

### LINKED ISSUE #742: Add check for different buckets having the same path in config
I just encountered an error whilst creating a test which was duplicate UUIDs in different stores. This was actually the result of a mistake in our `conftest.py` where two different stores have the same path. We should have some checks in place to make sure different stores don't have the same path as this can introduce confusing errors.

### PR DESCRIPTION
**Pull request by SutarPrasad**
* **Summary of changes** (Bug fix, feature, docs update, ...)
       Check's if the newly entered user input paths are present in the openghg.conf, And raises the value error

* **Please check if the PR fulfills these requirements**

- [x] Closes #742 (Replace xxxx with the Github issue number)
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature

**Not required**

- [ ] Documentation and tutorials updated/added
- [ ] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,10 @@ This version brings a breaking change with the move to use the [Zarr](https://za
 - Bug fix for allowing a period to be specified when this cannot be inferred from the input data [PR #899](https://github.com/openghg/openghg/pull/899)
 
 
+
+### Fixed
+- Added check's if the new paths provided by user are present in the object store and raises `ValueError`. - [PR #742](https://github.com/openghg/openghg/pull/904)
```
Comment (gareth-j) [inline CHANGELOG.md:43]:
```suggestion
- Added check for duplicate object store path being added under different store name, if detected raises `ValueError`. - [PR #742](https://github.com/openghg/openghg/pull/904)
```

File: openghg/util/_user.py [hunk 2]
```diff
@@ -126,9 +126,24 @@ def create_config(silent: bool = False) -> None:
 
         if new_shared_stores:
             existing = [k for k in new_shared_stores if k in stores]
+
+            existing_paths = [path["path"] for path in stores.values()]
```
Comment (gareth-j) [inline openghg/util/_user.py:130]:
```suggestion
            existing_paths = [path_data["path"] for path_data in stores.values()]
```

File: openghg/util/_user.py [hunk 3]
```diff
@@ -126,9 +126,24 @@ def create_config(silent: bool = False) -> None:
...
+
+            # Here it checks if newly entered paths are already present in config file.
+            duplicate_paths_with_store_name = {
+                store_name: path
+                for store_name, path in new_shared_stores.items()
+                if path["path"] in existing_paths
```
Comment (gareth-j) [inline openghg/util/_user.py:136]:
```suggestion
                if path_data["path"] in existing_paths
```

File: openghg/util/_user.py (hunk 2 above)
Comment (gareth-j) [reply openghg/util/_user.py:130]: I think `path_data` makes it clearer what's happening here.

**Review by gareth-j — CHANGES_REQUESTED** [review]
Just a few changes.

File: openghg/util/_user.py (hunk 2 above)
Comment (SutarPrasad) [reply openghg/util/_user.py]: done

File: openghg/util/_user.py (hunk 3 above)
Comment (SutarPrasad) [reply openghg/util/_user.py]: done

File: openghg/util/_user.py [hunk 4]
```diff
@@ -126,9 +126,24 @@ def create_config(silent: bool = False) -> None:
 
         if new_shared_stores:
             existing = [k for k in new_shared_stores if k in stores]
+
+            existing_paths = [path["path"] for path in stores.values()]
+
+            # Here it checks if newly entered paths are already present in config file.
+            duplicate_paths_with_store_name = {
+                store_name: path
+                for store_name, path in new_shared_stores.items()
```
[the remark this answers is not in the document]
Comment (SutarPrasad) [reply openghg/util/_user.py]: done

**Review by gareth-j — APPROVED** [review]
Looks good!

File: CHANGELOG.md [hunk 5]
```diff
@@ -42,6 +42,10 @@ This version brings a breaking change with the move to use the [Zarr](https://za
 - Bug fix for allowing a period to be specified when this cannot be inferred from the input data [PR #899](https://github.com/openghg/openghg/pull/899)
 
 
+
+### Fixed
```
Comment (rt17603) [inline CHANGELOG.md:46]: One note is that now we're released 0.8.0, this should be moved to the Unreleased part of the CHANGELOG.md file (i.e. between current lines 8 and 10)

Comment (SutarPrasad) [reply CHANGELOG.md:46]: Yes, Nice catch. Thanks for pointing it out.

Comment (SutarPrasad) [reply CHANGELOG.md]: done

**Review by gareth-j — DISMISSED** [review]
I think this is ready to go in.

**Pull request merged**